src/gpodder/util.py

```diff
@@ -1480,14 +1480,32 @@ def gui_open(filename):
     """
     try:
         if gpodder.ui.win32:
-            os.startfile(filename)
+            opener = None
         elif gpodder.ui.osx:
-            Popen(['open', filename], close_fds=True)
+            opener = 'open'
         else:
-            Popen(['xdg-open', filename], close_fds=True)
+            opener = 'xdg-open'
+
+        if opener is None:
+            os.startfile(filename)
+        else:
+            opener_fullpath = shutil.which(opener)
+            if opener_fullpath is None:
+                raise Exception((_("System default player '%(opener)s' not found"))
+                    % {'opener': opener}
+                )
+            Popen([opener_fullpath, filename], close_fds=True)
         return True
     except:
         logger.error('Cannot open file/folder: "%s"', filename, exc_info=True)
+        if gui is not None:
+            if opener is None:
+                message = _("Cannot open file/folder '%(filename)s' using default program") % {'filename': filename}
+            else:
+                message = _("Cannot open '%(filename)s' using '%(opener)s'") \
+                                        % {'filename': filename, 'opener': opener}
+                gui.show_message_details(_('Cannot open file/folder'),
```

Comment:
These two lines need to be dedented to handle both message branches.

Reply:
Of course, I somehow overlooked this.